src/components/Approval/Approval.tsx

```diff
@@ -157,6 +146,38 @@ export const Approval = () => {
     showErrorToast,
   ])
 
+  const toggleIsExactAllowance = useCallback(() => {
+    setIsExactAllowance(!isExactAllowance)
+  }, [isExactAllowance])
+
+  useEffect(() => {
+    setBuildApprovalTxInput(undefined)
+    createBuildApprovalTxInput(isExactAllowance)
+      .then(buildApprovalTxInput => {
+        setBuildApprovalTxInput(buildApprovalTxInput)
+      })
+      .catch(e => {
+        showErrorToast(e)
+        return history.push(TradeRoutePaths.Input)
+      })
```

Comment:
We conventionally don't use `.then().catch()` pipelines, with a few exceptions related to swallowing errors and still resolving to a value in case the promise throws
Very much nitpick and I'm happy either way, but since we're not awaiting this and consuming the resolved value (which would be `void` in both branches), we could make this a try/catch. Or alternatively just remove the return statement here and call it a day, either works 🙏🏽

Reply:
Also we conventionally use IIAFEs for async effects, which we would have to use if we move this to a try/catch as proposed